docs/cache_queries.py

```diff
@@ -170,10 +108,26 @@
     for label in ["home", "work"]
 ]
 
-for query in (
-    daily_location_queries
+print("Generating full dependency graph")
+dependency_graphs = [
+    flowmachine.utils.calculate_dependency_graph(q)
+    for q in daily_location_queries
     + modal_location_queries
-    + meaningful_locations_subqueries
     + meaningful_locations_queries
-):
-    query.store()
+]
+full_graph = dependency_graphs.pop()
+for graph in dependency_graphs:
+    full_graph.update(graph)
+
+print("Storing all queries and dependencies")
+all_query_stores = []
+for query in reversed(list(nx.topological_sort(full_graph))):
+    try:
+        all_query_stores.append(full_graph.nodes[query]["query_object"].store())
+    except ValueError:
+        # Some dependencies cannot be stored
+        pass
+
+
+print("Waiting for queries to finish")
+concurrent.futures.wait(all_query_stores)
```

Comment:
I'm wondering whether this still allows for queries to be stored in the "wrong" order (i.e., before all their dependencies have been stored), because the futures can potentially run in any order, no?

Reply:
Also, once this works we should probably add support for optionally storing all dependencies in `Query.store()` directly.

Reply:
The futures can run in any order, but the query state will be changed to `queued` before creating a future to store the query so queries should still be aware that their dependencies are going to be cached.

Reply:
Ah of course, excellent point. 👍